Objects/object.c

```diff
@@ -884,10 +885,31 @@ _PyObject_SetAttrId(PyObject *v, _Py_Identifier *name, PyObject *w)
     return result;
 }
 
+static inline int
+add_context_to_attribute_error_exception(PyObject* v, PyObject* name)
+{
+    assert(PyErr_Occurred());
+    // Intercept AttributeError exceptions and augment them to offer
+    // suggestions later.
+    if (PyErr_ExceptionMatches(PyExc_AttributeError)){
+        PyObject *type, *value, *traceback;
+        PyErr_Fetch(&type, &value, &traceback);
+        PyErr_NormalizeException(&type, &value, &traceback);
+        if (PyErr_GivenExceptionMatches(value, PyExc_AttributeError) &&
+            (PyObject_SetAttrString(value, "name", name) ||
+             PyObject_SetAttrString(value, "obj", v))) {
```

Comment:
Please use _Py_IDENTIFIER() API to avoid decoding byte strings from UTF-8 (and intern the Unicode objects) at each call.